text/0000-bee-model-crate/0000-bee-model-crate.md

```diff
@@ -0,0 +1,277 @@
+
+We also propose a `TransactionDraft` type that follows the builder pattern. It
+is intended to be an implementation detail and only be used by a `Bundle` builder
+type and not exposed publicly.
+
+# Motivation
+
+IOTA is a transaction settlement and data transfer layer for IoT (the Internet
+of Things). Messages on the network are `Bundle`s of individual `Transaction`s,
+which in turn are sent one at a time and are stored in a distributed ledger
+called the *Tangle*. Each `Transaction` encodes data such as sender and
+receiver addresses, reference nodes in the Tangle, `Bundle` hash, timestamps,
+and other information required to verify and process each transaction. With the
+transaction being the fundamental unit that is sent over the network, we need
+to represent it in memory.
+
+At the time of this RFC, the transaction format used by the IOTA network is
+defined by [release v1.8.1 of the IOTA Reference Implementation, IRI](`iri
+v1.8.1`), commit `e1776fbad5d90df86a26402f9025e4b0b2ef7d3e`. The transaction
+format might change in the future, but for the time being it is not yet
+understood how a common interface between different versions transaction
+formats should be implemented, or if the network will support several
+transaction types simultaneously.
+
+We thus do not consider generalizations over or interfaces for transactions,
+but only propose a basic `Transaction` type. All mentions of *transactions* in
+general or the `Transaction` type in particular will be implicitly in reference
+to that format used by `iri v1.8.1`.
+
+The `Transaction` type is intended to be *final* and immutable. This contract
+is enforced by not expose any methods that allow manipulating its fields
+directly. The only public constructor method that allows creating
+a `Transaction` directly are the `from_reader` and `from_slice` method to
+construct it from a type implementing `std::io::Read`, or from byte slice.
+Otherwise, `Transaction`s should only be built through `Bundle` constructors,
+and only from that context direct access to fields is permitted. To this
+purpose, this RFC contains the `TransactionDraft` builder pattern type.
+
+[`iri v1.8.1`]: https://github.com/iotaledger/iri/releases/tag/v1.8.1-RELEASE
+
+# Detailed design
+
+A transaction is a sequence of 15 fields of constant length. A transaction has
+a total length of 8019 trits. The fields' name, description, and size (in
+trits) is summarized in the table below, in their order of appearance in
+a transaction.
+
+| Name                              | Description                                            | Size (in trits) |
+| ---                               | ---                                                    | ---             |
+| `signature_or_message_fragment`   | contains the signature of the transfer                 |                 |
+|                                   | or user-defined message data                           | 6561            |
```

Comment:
I think it's no need to be separate line. Besides, this would become totally different row. Normal markdown display would just adjust them.

```suggestion
| `signature_or_message_fragment` | contains the signature of the transfer or user-defined message data | 6561 |
```

Reply:
@SuperFluffy I think we could ignore max chars limit during table format. We should make representation correct at least.